src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+                modifiedMembers.Add(member);
+            }
+
+            if (equals != null && notEquals != null)
+            {
+                if (IsDefaultEqualsOperator(equals, type, semanticModel, cancellationToken) &&
+                    IsDefaultNotEqualsOperator(notEquals, type, semanticModel, cancellationToken))
+                {
+                    modifiedMembers.Remove(equals);
+                    modifiedMembers.Remove(notEquals);
+                }
+            }
+
+            return modifiedMembers.ToImmutable();
+        }
+
+        /// <summary>
+        /// Returns true if the method contents match a simple reference to the equals method
+        /// which would be the compiler generated implementation
+        /// </summary>
+        private static bool IsDefaultEqualsOperator(
+            OperatorDeclarationSyntax operatorDeclaration,
+            INamedTypeSymbol type,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            if (semanticModel.GetOperation(operatorDeclaration, cancellationToken) is not IMethodBodyOperation operation)
+            {
+                return false;
+            }
+
+            var body = operation.BlockBody ?? operation.ExpressionBody;
+
+            // must look like
+            // public static operator ==(C c1, object? c2)
+            // {
+            //  return c1.Equals(c2);
+            // }
+            // or
+            // public static operator ==(C c1, object? c2) => c1.Equals(c2);
+            // can have either object or class type as type for second param
+            return body != null &&
+                body.Operations.IsSingle() &&
+                operation.BlockBody?.Operations[0] is IReturnOperation returnOperation &&
+                returnOperation.ChildOperations.Count == 1 &&
+                IsDotEqualsInvocation(returnOperation.ChildOperations.First(), type);
+        }
+
+        // must be of form !(c1 == c2) or !c1.Equals(c2)
+        private static bool IsDefaultNotEqualsOperator(
+            OperatorDeclarationSyntax operatorDeclaration,
+            INamedTypeSymbol type,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            if (semanticModel.GetOperation(operatorDeclaration, cancellationToken) is not IMethodBodyOperation operation)
+            {
+                return false;
+            }
+
+            var body = operation.BlockBody ?? operation.ExpressionBody;
+            // if any of these conditions are false we short-circuit false
+            if (!(body != null &&
+                body.Operations.IsSingle() &&
+                operation.BlockBody?.Operations[0] is IReturnOperation returnOperation &&
+                returnOperation.ChildOperations.Count == 1 &&
+                returnOperation.ChildOperations.First() is IUnaryOperation notOp &&
+                notOp.Syntax.IsKind(SyntaxKind.LogicalNotExpression) &&
```

Comment:
avoid going back to syntax from operations.  the operation shoudl likely already contain this info, probably as some sort of OperatorKind field.